src/query/split/split_trimesh.rs

```diff
@@ -344,6 +346,274 @@ impl TriMesh {
+                        // Intersect the segment with the plane.
+                        if let Some((intersection, _)) = segment
+                            .local_split_and_get_intersection(local_axis, bias, epsilon)
+                            .1
+                        {
+                            new_vertices.push(intersection);
+                            colors.push(0);
+                            (new_vertices.len() - 1) as u32
+                        } else {
+                            unreachable!()
+                        }
+                    })
+            };
+
+            // Perform the intersection, push new triangles, and update
+            // triangulation constraints if needed.
+            match intersection_features {
+                (_, FeatureId::Unknown) => {
+                    // The plane doesn’t intersect the triangle, or intersects it at
+                    // a single vertex, so we don’t have anything to do.
+                    assert!(
+                        matches!(intersection_features.0, FeatureId::Unknown)
+                            || matches!(intersection_features.0, FeatureId::Vertex(_))
+                    );
+                }
+                (FeatureId::Vertex(iv1), FeatureId::Vertex(iv2)) => {
+                    // The plane intersects the triangle along one of its edge.
+                    // We don’t have to split the triangle, but we need to add
+                    // the edge to the polyline indices
+
+                    let id1 = idx[iv1 as usize];
+                    let id2 = idx[iv2 as usize];
+
+                    let out_id1 = *existing_vertices_found.entry(id1).or_insert_with(|| {
+                        let v1 = vertices[id1 as usize];
+
+                        new_vertices.push(v1);
+                        (new_vertices.len() - 1) as u32
+                    });
+                    let out_id2 = *existing_vertices_found.entry(id2).or_insert_with(|| {
+                        let v2 = vertices[id2 as usize];
+
+                        new_vertices.push(v2);
+                        (new_vertices.len() - 1) as u32
+                    });
+
+                    add_segment_adjacencies(out_id1, out_id2);
+                }
+                (FeatureId::Vertex(iv), FeatureId::Edge(ie))
+                | (FeatureId::Edge(ie), FeatureId::Vertex(iv)) => {
+                    // The plane splits the triangle into exactly two triangles.
+                    let ia = ie;
+                    let ib = (ie + 1) % 3;
+                    let ic = (ie + 2) % 3;
+                    let idx_a = idx[ia as usize];
+                    let idx_b = idx[ib as usize];
+                    let idx_c = idx[ic as usize];
+                    assert_eq!(iv, ic);
+
+                    let intersection_idx = intersect_edge(idx_a, idx_b);
+
+                    add_segment_adjacencies(idx_c, intersection_idx);
```

Comment:
That `idx_c` looks incorrect? It belongs to the mesh’s index space instead of indexing into the `new_vertices` array. Unless I missed something, I would expect and index conversion like in [the vertex/vertex case](https://github.com/dimforge/parry/pull/114/files#diff-69c40d9287d4877ea22122e7d1fa7509c093a5e0edb3d77571bb8116a5df2ff6R520-R525) to happen for `idx_c`.

Reply:
I think you're right. Good catch! I'll add a test for this case as well.

Reply:
Fixed as of b7693f042c3509986e10926a225cd871ad9d445b